zeroex/orderwatch/order_watcher.go

```diff
@@ -264,20 +267,31 @@ func (w *Watcher) cleanupLoop(ctx context.Context) error {
 }
 
 func (w *Watcher) removedCheckerLoop(ctx context.Context) error {
-	for {
-		start := time.Now()
-		if err := w.permanentlyDeleteStaleRemovedOrders(ctx); err != nil {
-			return err
-		}
+	lastDeleted := time.Now()
+	lastCounted := time.Now()
 
+	for {
 		select {
 		case <-ctx.Done():
 			return nil
-		// Wait minRemovedCheckInterval before calling permanentlyDeleteStaleRemovedOrders again. Since
-		// we only start waiting _after_ permanentlyDeleteStaleRemovedOrders completes, we will never
-		// have multiple calls to permanentlyDeleteStaleRemovedOrders running in parallel
-		case <-time.After(minRemovedCheckInterval - time.Since(start)):
-			continue
+		case <-time.After(maxDeleteInterval - time.Since(lastDeleted)):
```

Comment:
One effect of writing the `for` loop this way is that when _either_ `case` is triggered it causes us to break out of the `select` statement and continue to the next iteration of the `for` loop. This causes us to call `time.After` twice more, which would result in unnecessarily creating two new channels. Can we coalesce this into a single call to `time.After` instead? This would cut calls to `time.After` (and therefore the number of new channels created) in half.

For example, something like this:

```go
for {
	select {
	case <-ctx.Done():
		return nil
	case <-time.After(countInterval - time.Since(lastCounted)):
		if time.Since(lastDeleted) > maxDeleteInterval {
			// If it has been longer than maxDeleteInterval since the last time
			// we deleted stale orders, go ahead and delete them now.
			if err := w.permanentlyDeleteStaleRemovedOrders(ctx); err != nil {
				return err
			}
			lastDeleted = time.Now()
		} else {
			// Otherwise, check the current number of orders in the dataase to see
			// if we are above databaseUtilizationThreshold.
			count, err := w.db.CountOrders(nil)
			if err != nil {
				return err
			}
			lastCounted = time.Now()

			if float64(count)/float64(w.maxOrders) > databaseUtilizationThreshold {
				if err := w.permanentlyDeleteStaleRemovedOrders(ctx); err != nil {
					return err
				}
				lastDeleted = time.Now()
			}
		}
	}
}
```

Technically this means we could be waiting up to `countInterval` longer than `maxDeleteInterval` between calls to `permanentlyDeleteStaleRemovedOrders`, but I don't think that's really an issue.